tests/unit_tests/dft/include/compute_out_of_place.hpp

```diff
@@ -0,0 +1,143 @@
+
+#ifndef ONEMKL_COMPUTE_OUT_OF_PLACE_HPP
+#define ONEMKL_COMPUTE_OUT_OF_PLACE_HPP
+
+/* Note: There is no implementation for Domain Real */
+template <dft::precision precision, dft::domain domain>
+int DFT_Test<precision, domain>::test_out_of_place_buffer() {
+    if (!init()) {
+        return test_skipped;
+    }
+
+    descriptor_t descriptor{ size };
+    descriptor.set_value(oneapi::mkl::dft::config_param::PLACEMENT,
+                         oneapi::mkl::dft::config_value::NOT_INPLACE);
+    commit_descriptor(descriptor, sycl_queue);
+
+    std::vector<OutputType> out_host(size);
+    std::vector<InputType> out_host_back(size);
+    sycl::buffer<InputType, 1> in_dev{ sycl::range<1>(size) };
+    sycl::buffer<OutputType, 1> out_dev{ sycl::range<1>(size) };
+    sycl::buffer<InputType, 1> out_back_dev{ sycl::range<1>(size) };
+
+    copy_to_device(sycl_queue, input, in_dev);
+
+    try {
+        dft::compute_forward<std::remove_reference_t<decltype(descriptor)>, InputType, OutputType>(
+            descriptor, in_dev, out_dev);
+    }
+    catch (oneapi::mkl::unimplemented &e) {
+        std::cout << "Skipping test because: \"" << e.what() << "\"" << std::endl;
+        return test_skipped;
+    }
+
+    copy_to_host(sycl_queue, out_dev, out_host);
+
+    EXPECT_TRUE(check_equal_vector(out_host.data(), out_host_ref.data(), out_host.size(), 1,
+                                   error_margin, std::cout));
+
+    descriptor_t descriptor_back{ size };
+    descriptor_back.set_value(oneapi::mkl::dft::config_param::PLACEMENT,
+                              oneapi::mkl::dft::config_value::NOT_INPLACE);
+    descriptor_back.set_value(dft::config_param::BACKWARD_SCALE, (1.0 / size));
+    commit_descriptor(descriptor_back, sycl_queue);
+
+    try {
+        dft::compute_backward<std::remove_reference_t<decltype(descriptor_back)>, OutputType,
+                              InputType>(descriptor_back, out_dev, out_back_dev);
+    }
+    catch (oneapi::mkl::unimplemented &e) {
+        std::cout << "Skipping test because: \"" << e.what() << "\"" << std::endl;
+        return test_skipped;
+    }
+
+    copy_to_host(sycl_queue, out_back_dev, out_host_back);
+
+    EXPECT_TRUE(check_equal_vector(out_host_back.data(), input.data(), input.size(), 1,
+                                   error_margin, std::cout));
+
+    return !::testing::Test::HasFailure();
+}
+
+template <dft::precision precision, dft::domain domain>
+int DFT_Test<precision, domain>::test_out_of_place_USM() {
+    if (!init()) {
+        return test_skipped;
+    }
+
+    descriptor_t descriptor{ size };
+    descriptor.set_value(oneapi::mkl::dft::config_param::PLACEMENT,
+                         oneapi::mkl::dft::config_value::NOT_INPLACE);
+    commit_descriptor(descriptor, sycl_queue);
+
+    std::vector<OutputType> out_host(size);
+    std::vector<InputType> out_host_back(size);
+
+    auto ua_input = usm_allocator<InputType, usm::alloc::shared, 64>(cxt, *dev);
```

Comment:
I think there could be an usm_allocator_t somewhere since you use it a lot:
```
template <typename ElemT>
using usm_allocator_t = usm_allocator<ElemT, usm::alloc:shared, 64>;
```